shared/msg_conversions/tools/pose_stamped_msg_cnv.cc

```diff
@@ -48,12 +60,12 @@ void odometry_callback(geometry_msgs::PoseStampedPtr const & odometry) {
 
 int main(int argc, char** argv) {
   ff_common::InitFreeFlyerApplication(&argc, &argv);
-  ros::init(argc, argv, "pose_stamped_msg_cnv");
+  ROS_CREATE_NODE("pose_stamped_msg_cnv");
+
 
-  ros::NodeHandle nh;
-  ros::Subscriber odometry_sub = nh.subscribe(FLAGS_input_topic, 5, &odometry_callback);
-  transform_pub.reset(new tf2_ros::TransformBroadcaster());
-  ros::spin();
+  auto odometry_sub = ROS_CREATE_SUBSCRIBER(geometry_msgs::PoseStamped, FLAGS_input_topic, 5, odometry_callback);
+  // transform_pub.reset(new tf2_ros::TransformBroadcaster());
```

Comment:
not sure why this line is necessary and why it breaks the build on ROS2

Reply:
In ROS2 `TransformBroadcaster()` takes a required node argument.

This could be abstracted with a function `ff_ros::tf2_ros::get_transform_broadcaster(node)` that calls the `tf2_ros::TransformBroadcaster()` constructor with or without the node argument depending on ROS version. Or we could just write out the conditional each time if it's not called too often.

Reply:
(Without the call `transform_pub` is basically a null pointer so any attempt to use it should crash.)